jabra/callstats-jabra-shim.js

```diff
@@ -0,0 +1,192 @@
+        CallstatsJabraShim.conferenceID, {type: eventType});
+    }
+
+    function initJabraEventListeners() {
+      jabra.addEventListener("device attached", () => {
+        console.log("CallstatsJabraShim: device attached");
+      });
+
+      jabra.addEventListener("mute", (event) => {
+        sendCustomEvent('mute');
+      });
+
+      jabra.addEventListener("unmute", (event) => {
+        sendCustomEvent('unmute');
+      });
+
+      jabra.addEventListener("devlog", (event) => {
+        let noiseDb = undefined;
+        let txLevelEvent = event.data["TX Acoustic Logging Level"];
+        if (txLevelEvent !== undefined) {
+          noiseDb = parseInt(txLevelEvent);
+        }
+
+        let exposureDb = undefined;
+        let rxLevelEvent = event.data["RX Acoustic Logging Level"];
+        if (rxLevelEvent !== undefined) {
+          exposureDb = parseInt(rxLevelEvent);
+        }
+        sendCustomStats(noiseDb, exposureDb);
+
+        let muteState = event.data["Mute State"];
+        if (muteState === "TRUE") {
+          sendCustomEvent('mute');
+        } else if (muteState === "FALSE") {
+          sendCustomEvent('unmute');
+        }
+        
+        let boomArmEvent = event.data["Boom Position Guidance OK"];
+        if (boomArmEvent === "TRUE") {
+          sendCustomEvent('boomArmPositionOk');
+        } else if (boomArmEvent === "FALSE") {
+          sendCustomEvent('boomArmPositionNotOk');
+        }
+
+        let txSpeech = undefined;
+        let txSpeechEvent = event.data["Speech_Analysis_TX"];
+        if (txSpeechEvent !== undefined) {
+            txSpeech = (txSpeechEvent.toString().toLowerCase() === "true");
+        }
+
+        let rxSpeech = undefined;
+        let rxSpeechEvent = event.data["Speech_Analysis_RX"];
+        if (rxSpeechEvent !== undefined) {
+            rxSpeech = (rxSpeechEvent.toString().toLowerCase() === "true");
+        }
+
+        sendJabraActiveState(txSpeech, rxSpeech);
+      });
+    }
+
+    function initJabraLib() {
+      // Jabra library init with full installation check, focus setup and diagnostics of common problems:
+      jabra.init().then(() => jabra.getInstallInfo()).then((installInfo) => {
+        console.log("CallstatsJabraShim: Jabra library initialized");
+        if (installInfo.installationOk) {
+          console.log("CallstatsJabraShim: Jabra Installation check succeded");
+          initJabraEventListeners();
+        } else {
+          console.error("CallstatsJabraShim: Browser SDK Installation incomplete. Please (re)install");
+        }
+      }).catch((err) => {
+          if (err.name === "CommandError" && err.errmessage === "Unknown cmd" && err.command === "getinstallinfo" ) {
+            console.log("CallstatsJabraShim: Your browser SDK installation is incomplete, out of date or corrupted. Please (re)install");
+          } else if (err.name === "NotFoundError") {
+            console.log("CallstatsJabraShim: Input device not accessible/found");
```

Comment:
Should we send an error event here and in other places where potential errors can happen?